### PR TITLE
Logging library

### PR DESCRIPTION
**Pull request by StRhys**
This is the same ticket as this one for the [homepage](https://github.com/GSS-Cogs/idpd-frontend-homepage/pull/56).

This ticket was to improve how our logger works.

Using [next-logger](https://github.com/sainsburys-tech/next-logger) to output logs in json, which uses [pino](https://github.com/pinojs/pino) to format and add fields to the output. Using the next-logger.config to configure our logs to conform to [dp logging standards](https://github.com/ONSdigital/dp-standards/blob/main/LOGGING_STANDARDS.md).
It's setup so full needed log is only done in production and then we use [pino-pretty](https://github.com/pinojs/pino-pretty) in dev so we have more readable outputs in the terminal.

to test
do yarn dev go on the app, should see some nice looking logs in terminal.
Add some of your own console.info and console.error to see different outputs.

do make build and make start you should now see the output in json format in terminal.

### CODE REVIEW

File: libs/dataRequests.tsx [hunk 1]
```diff
@@ -1,17 +1,50 @@
...
+      host: domain,
+      // port: // TODO add port number
+      path: url,
+      started_at: moment(new Date()).format("YYYY-MM-DDTHH:mm:ss.SSSZZ"),
+    },
+  });
+};
+
+const logError = (message: string, method: string, url: string) => {
+  console.error({
```
Comment (mikeAdamss) [inline libs/dataRequests.tsx:30]: is there any version of a stack trace in frontend land or would that be what we're passing in as the message?

if its in the message I'd probably use that to populate the error field and use a simple english message for event, i.e

```
message: "Rhys saying whats gone wrong where and when in an idiot friendly way, i.e failed to get x`
error: [whatever error we're getting back - the more raw message from the system for whoever needs to figure out the issue]
```

does that make sense?

File: libs/dataRequests.tsx [hunk 2]
```diff
@@ -1,17 +1,50 @@
...
+const domain = backendUrlSplit?.[1];
+
+const logInfo = (message: string, method: string, url: string) => {
+  console.info({
+    event: message,
+    http: {
+      method: method,
+      scheme: scheme,
+      host: domain,
+      // port: // TODO add port number
```
Comment (mikeAdamss) [inline libs/dataRequests.tsx]: TODO needs doing 😄 , there's a few of these.

**Review by mikeAdamss — COMMENTED** [review]
minor comments/questions

File: libs/dataRequests.tsx (hunk 1 above)
Comment (StRhys) [reply libs/dataRequests.tsx:30]: Swtiched up how tthe errors are handled, so we can grab the stack. Following the dp standards, so it'll be `errors: [{our error}}` will output tto something similar as below.
image

File: libs/dataRequests.tsx (hunk 2 above)
Comment (StRhys) [reply libs/dataRequests.tsx]: Added a `BACKEND_PORT` env var as a means to get the port as unable to do it frontside. So will need to be added at some point. Currently defaults to blank.

**Comment by mikeAdamss** [comment]
closing as prioritise have changed, may reopen later.

**Pull request closed without merging**